R/featEng-entropy.R

```diff
@@ -0,0 +1,58 @@
+#' @title Computing Region Entropy
+#'
+#' @description Computes the entropy of the MRI voxel values for each patient
+#' stratified on individual clusters.
+#'
+#' @details \code{computeRegionEntropy()} takes in two arguments: a \code{tibble}
+#' voxel_df and a \code{tibble} data_df. \code{voxel_df} is assumed to be in long
+#' format, with each individual pixel being represented on a single row, so that
+#' the first columns is 'pid,' columns 2 to 4 are 'x', 'y', 'z', and column 5
+#' is 'value'. \code{data_df} is assumed to be the output of \code{dataDrivenClusters()}.
+#' \code{computeRegionEntropy()} combines \code{voxel_df} and \code{data_df} into
+#' a single \code{tibble}, and then pivots this combined data with respect to
+#' the columns 'pid' and 'cluster'. We then find the entropy of each pid-cluster
+#' pair. If \code{data_df} is \code{null}, then the function assumes that all
+#' the observations come from the same cluster.
+#'
+#' @param voxel_df A \code{dataframe} in the following format: the first column
+#' is assumed to be an index column counting from 1 to \code{nrow(voxel_df)},
+#' and the second column is a column labeled "PID," which contains patient
+#' identification. The third, fourth, and fifth columns are 'x', 'y', and 'z'
```

Comment:
We should not have the user order columns this way. The code should run as long as the column names are correctly specified, not the names AND the order. The user should do as little work as possible